### PR TITLE
Add naiad-commons recipe

### PR DESCRIPTION
**Pull request by Marie59**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Marie59** [comment]
Hi @conda-forge/help-python this recipe is ready ! Thanks !

**Comment by Marie59** [comment]
Hi @conda-forge/help-python @conda-forge/staged-recipes can anyone review my recipe ? I need it for the naiad recipe in waiting for the moment.

File: recipes/naiad-commons/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{% set name = "naiad-commons" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://gitlab.ifremer.fr/naiad/naiad-commons.git
```
Comment (ocefpaf) [inline recipes/naiad-commons/meta.yaml:9]:
```suggestion
  url: https://gitlab.ifremer.fr/naiad/naiad-commons/-/archive/{{ version }}/naiad-commons-{{ version }}.tar.gz
```

File: recipes/naiad-commons/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
+
+requirements:
+  host:
+    - python >=3.9
+    - pip
+    - wheel
+    - poetry
```
Comment (ocefpaf) [inline recipes/naiad-commons/meta.yaml:22]:
```suggestion
    - poetry-core
```

File: recipes/naiad-commons/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - poetry-dynamic-versioning
+
+  run:
+    - python >=3.9
+    - scipy
```
Comment (ocefpaf) [inline recipes/naiad-commons/meta.yaml:27]:
```suggestion
    - scipy >=1.9.2
```

File: recipes/naiad-commons/meta.yaml (hunk 1 above)
Comment (Marie59) [reply recipes/naiad-commons/meta.yaml:9]: With the use of poetry-dynamic-versioning the url does not work it seems to need the .git url. Is it really not possible to use the git_url ?

Comment (ocefpaf) [reply recipes/naiad-commons/meta.yaml:9]: See https://github.com/conda-forge/staged-recipes/pull/21655#discussion_r1070989128

PS: For simplicity let's keep the discussion there too and, when we resolve, we can apply the solution here as well.

**Comment by ocefpaf** [comment]
Thanks @maresb, @xylar and @BastianZim for the assist!
Thanks @Marie59 for the quick fix and the recipe!

**Pull request merged**